### PR TITLE
chore: add check for blockers before doing...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

chore: add check for blockers before doing release (#20672); note: works locally, not tested in GHA environment yet (the ubuntu stuff)

Change-Id: I3114ac445d547bd464e9d87608a4f0776d7f072a
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20672

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: make-release.sh [hunk 1]
```diff
@@ -28,6 +28,32 @@ usage ()
...
+    # check for blockers only if doing a 7.yy.0 release
+    if [[ ${CHE_VERSION} == *".0" ]]; then 
+        # install gh cli from https://github.com/cli/cli/releases/
+        sudo dpkg -i https://github.com/cli/cli/releases/download/v2.2.0/gh_2.2.0_linux_amd64.deb
+
+        # find blockers for a given milestone
+        # BLOCKERS_THIS_MILESTONE="$(gh issue list -R eclipse/che -l "severity/blocker" -s "open" -m "${CHE_VERSION%.*}" --json "createdAt,updatedAt,author,title,url,milestone" | jq -r '.[]')"
+
+        # find any blockers, including those unassigned to milestones
+        BLOCKERS_ANY="$(gh issue list -R eclipse/che -l "severity/blocker" -s "open" --json "createdAt,updatedAt,author,title,url,milestone" | jq -r '.[]')"
```
Comment (benoitf) [inline make-release.sh:42]: maybe it's simpler to call the API directly ?
```
$ curl -s "https://api.github.com/repos/eclipse/che/issues?labels=severity/blocker" | jq -r '.'
```

Comment (nickboldt) [reply make-release.sh:42]: simpler perhaps. But how can I pass in status (open/closed) or milestone to that query?

Comment (benoitf) [reply make-release.sh:42]: By default the api is returning only open issues

But you can add ?state=closed for example to get closed issues or state=open

The syntax is longer than the direct API call :-)

Comment (benoitf) [reply make-release.sh:42]: And for your request you're only interested in the array length so I don't know why you might want to select fields anyway

File: make-release.sh [hunk 2]
```diff
@@ -28,6 +28,32 @@ usage ()
   exit 1
 }
 
+checkForBlockerIssues()
+{
+    # check for blockers only if doing a 7.yy.0 release
+    if [[ ${CHE_VERSION} == *".0" ]]; then 
+        # install gh cli from https://github.com/cli/cli/releases/
```
Comment (benoitf) [inline make-release.sh:35]: I would not use cli for such simple request but anyway this cli is already installed on runners

File: make-release.sh (hunk 1 above)
Comment (nickboldt) [reply make-release.sh:42]: The idea was to provide useful information to the person who has to then chase down the blocker issues.

```
milestone="7.39"
milestoneID=$(curl -s "https://api.github.com/repos/eclipse/che/milestones?sort_on=due_on&direction=desc&state=open" | jq -r --arg milestone $milestone '.[]|select(.title==$milestone)|.number' 2>&1) # 151

curl -s "https://api.github.com/repos/eclipse/che/issues?labels=severity/blocker&state=closed&milestone=${milestoneID}" | jq -r '.[]|[.created_at,.updated_at,.milestone.title,.url,.user.login,.title] | @tsv'

2021-11-02T13:53:01Z	2021-11-03T11:50:14Z	7.39	https://api.github.com/repos/eclipse/che/issues/20716	vinokurig	Che-theia editor docker image build fails
2021-10-18T08:19:51Z	2021-11-03T11:44:38Z	7.39	https://api.github.com/repos/eclipse/che/issues/20644	sleshchenko	Che should supply devfile v2 in Che Server mode (by converting it to devfile v1)
```

**Review by benoitf — APPROVED** [review]

**Comment by nickboldt** [comment]
And... it works!

image

https://github.com/eclipse-che/che-release/runs/4122608509?check_suite_focus=true

**Comment by benoitf** [comment]
Maybe it should print the HTML url instead of the api url

**Comment by nickboldt** [comment]
open to submissions as PRs :)

**Pull request merged**